src/libraries/System.Diagnostics.Process/src/System/Diagnostics/Process.BSD.cs

```diff
@@ -18,27 +18,24 @@ public partial class Process
         [SupportedOSPlatform("maccatalyst")]
         public static Process[] GetProcessesByName(string? processName, string machineName)
         {
-            if (processName == null)
-            {
-                processName = string.Empty;
-            }
+            ProcessManager.ThrowIfRemoteMachine(machineName);
 
-            Process[] procs = GetProcesses(machineName);
-            var list = new List<Process>();
+            int[] procIds = ProcessManager.GetProcessIds();
+            var processes = string.IsNullOrEmpty(processName) ?
+                new List<Process>(procIds.Length) :
+                new List<Process>();
```

Comment:
What about always allocating Process[procIds.Length] and resize when there is null entry?

Reply:
`ArrayBuilder` under Common does exactly that.